src/apps/relay/dbdrivers/dbdriver.c

```diff
@@ -45,7 +45,7 @@ static void make_connection_key(void) { (void)pthread_key_create(&connection_key
 pthread_key_t connection_key;
 pthread_once_t connection_key_once = PTHREAD_ONCE_INIT;
 
-int convert_string_key_to_binary(char *keysource, hmackey_t key, size_t sz) {
+void convert_string_key_to_binary(char const *keysource, hmackey_t key, size_t sz) {
```

Comment:
`convert_string_key_to_binary()` never returned anything but 0.
Changed the function to return void to simplify logic at call-sites.

Also, changed input parameter to `char const*` to allow the optimizer to assume that the contents of `keysource` won't be modified.